### DISCUSSION TITLE
AUR ddns-updater package for Arch Linux

### DISCUSSION THREAD

**Discussion by Marcool04** [qdm12/ddns-updater#808] [Show and tell]
Hi there. Thanks for the great little tool that is ddns-updater. I just wanted to let people here know that I've adopted the AUR package https://aur.archlinux.org/packages/ddns-updater, updated it to the latest release, and improved the package a bit, adding a service file and such.

Please let me know if anybody sees a problem with it.

Best,

Mark.

**Comment by qdm12** [comment]
Hi Mark, that's great, thanks for sharing!
If you need/would prefer to have some files such as the service file directly in this repository, feel free to let me know.
I've added a link to the AUR package in the readme in a094476345d888c3d55e8a48d44667306aa58a95

**Reply by Marcool04** [reply]
Thanks for that! That's great.
I don't mind either way about the service. If you feel it might be useful for others to have it in the repo here that's fine by me, otherwise I am used to including service files in AUR packages, they are often not included but I find them very useful so write my own in most cases.
Feel free to include something along the lines of what I wrote:

```
[Unit]
Description=Dynamic DNS update daemon
After=network-online.target
Wants=network-online.target

[Service]
SyslogIdentifier=ddns-updater
WorkingDirectory=/var/lib/ddns-updater
User=ddns-updater
Environment=DATADIR=/var/lib/ddns-updater/data
ExecStart=/usr/bin/ddns-updater 

[Install]
WantedBy=multi-user.target
```

This could certainly be improved with hardening and other tweaks. Note that it does assume a  `ddns-updater` user has been created and has its home dir in `/var/lib`.

**Comment by hjheins** [comment]
Hi,

I am using the arch package of the updater. However, I have some environment variables I would like to use.
How can I do so except for editing the systemd file?

thanks

**Reply by Marcool04** [reply]
Hi @hjheins.

That's a good point. I hadn't really looked into the issue when I created the package. I used a hardcoded `DATADIR=/var/lib/ddns-updater/data` because I felt there wasn't much else needed changing, but as there are indeed quite a few [environment variables](https://github.com/qdm12/ddns-updater/blob/master/README.md#environment-variables) available, I have reworked the PKGBUILD to include an environment file that is sourced in a (modified) systemd service file. Please try updating to version 2.8.1-2 and let me know if anything seems off to you.

Best regards,

Mark.

**Comment by hjheins** [comment]
@Marcool04 , thanks, that seems to work. :-)
By the way: I am running on an arm system. so you can add arm7h to the supported architectures ;-)

cheers,

Hendrik-Jan

**Reply by Marcool04** [reply]
Yep, seems fair. Done with the 2.9.0 version
